gap/OscarInterface/gap/alnuth.gi

```diff
@@ -0,0 +1,171 @@
+end);
+
+BindGlobal("UnitGroupDescriptionOscar", function(F)
+  local K, O, U_m, U, m, basis;
+
+  K := _OscarField(F);
+  O := Oscar.maximal_order(K);
+  U_m := Oscar.unit_group(O);
+  U := U_m[1];
+  m := U_m[2];
+
+  basis := Julia.map(Oscar.coordinates, Julia.map(K, Julia.map(m, Oscar.gens(U))));
+
+  return JuliaToGAP(IsList, basis, true);
+end);
+
+BindGlobal("ExponentsOfUnitsDescriptionWithRankOscar", function(F, elms)
+  local K, O, U_m, U, m, basis, units, rank, expns, x;
+
+  K := _OscarField(F);
+  O := Oscar.maximal_order(K);
+  U_m := Oscar.unit_group(O);
+  U := U_m[1];
+  m := U_m[2];
+
+  basis := Julia.map(Oscar.coordinates, Julia.map(K, Julia.map(m, Oscar.gens(U))));
+  units := JuliaToGAP(IsList, basis, true);
+
+  # the order of the torsion part of the full unit group
+  rank := Oscar.GAP.julia_to_gap(Oscar.order(U_m[1][1]));
+
+  expns := [];
+  for x in elms do
+     # map GAP int vector to Oscar element
+     Add(expns, Oscar.GAP.julia_to_gap(Oscar.preimage(m, K(GAPToJulia(x))).coeff)[1]);
+  od;
+
+  # return result
+  return rec(units := units, expns := expns, rank := rank);
+
+end);
+
+BindGlobal("ExponentsOfFractionalIdealDescriptionOscar", function(F, elms)
+  local K, O, tmp, ideals, result, blah, vec, I;
+
+  K := _OscarField(F);
+  O := Oscar.maximal_order(K);
+
+  tmp := Julia.map(x -> Oscar.factor(K(GAPToJulia(x)) * O), elms);
+  # take the union of the keys
+  ideals := Julia.collect(Julia.reduce(Oscar.union, Julia.map(Oscar.Set, Julia.map(Oscar.keys, tmp))));
+  if Julia.isempty(ideals) then
+    return [];
+  fi;
+
+  result := [];
+  for blah in JuliaToGAP(IsList, tmp) do
+    vec := [];
+    for I in JuliaToGAP(IsList, ideals) do
+      Add(vec, Julia.get(blah, I, 0));
+    od;
+    Add(result, vec);
+  od;
+
+  return result;
+end);
+
+BindGlobal("NormCosetsDescriptionOscar", function(F, norm)
+  local K, O, U_m, U, m, basis, units, eqn, creps;
+
+  K := _OscarField(F);
+  O := Oscar.maximal_order(K);
+  U_m := Oscar.unit_group(O);
+  U := U_m[1];
+  m := U_m[2];
+
+  basis := Julia.map(Oscar.coordinates, Julia.map(K, Julia.map(m, Oscar.gens(U))));
+  units := JuliaToGAP(IsList, basis, true);
+
+  eqn := Oscar.norm_equation(O, norm);
+
+  creps := JuliaToGAP(IsList, Julia.map(Oscar.coordinates, Julia.map(K, eqn)), true);
+
+  # return result
+  return rec(units := units, creps := creps);
+end);
+
+
+
+BindGlobal("PolynomialFactorsDescriptionOscar", function(F, coeffs)
+  local K, cf, poly, facs, result, f, g, i;
+
+  K := _OscarField(F);
+
+  cf := JuliaEvalString( "Vector{nf_elem}")(Reversed(List(coeffs, x -> K(GAPToJulia(x)))));
+  poly := Oscar.polynomial(K, cf);
+  facs := Oscar.factor(poly);
+  Assert(0, Oscar.is_one(facs.unit));
+
+  result := [];
+  for f in JuliaToGAP(IsList, Oscar.collect(facs.fac)) do
+    # Convert factor to GAP
+    g := Julia.map(Oscar.coordinates, f[1].coeffs);
+    g := CallJuliaFunctionWithKeywordArguments(Oscar.GAP.julia_to_gap, [g], rec( recursive := true));
```

Comment:
Looks complicated. What is wrong with `JuliaToGAP(IsList, g, true)`?

Reply:
Huh, good question. I am pretty sure I tried that and there was an error back then, but I probably just did it wrong somehow... It definitely passes the tests now, so I'll use that. Thanks!